cilium/cmd/monitor.go

```diff
@@ -59,40 +63,97 @@ var (
 		SampleType:   bpf.PERF_SAMPLE_RAW,
 		WakeupEvents: 1,
 	}
+	eventTypeIdx = -1 // for integer comparison
+	eventType    = ""
+	eventTypes   = []string{"drop", "debug", "capture"}
+	fromSource   = 0
+	toDst        = 0
 )
 
 func lostEvent(lost *bpf.PerfEventLost, cpu int) {
 	fmt.Printf("Lost %d events\n", lost.Lost)
 }
 
-func receiveEvent(msg *bpf.PerfEventSample, cpu int) {
-	prefix := fmt.Sprintf("CPU %02d:", cpu)
+// match checks if the event type, from endpoint and to endpoint match when they
+// are supplied. If either one of them is less than zero, then it is assumed user did
+// use them.
+func match(messageType int, src uint16, dst uint32) bool {
+	if eventTypeIdx > bpfdebug.MessageTypeUnspec && messageType != eventTypeIdx {
```

Comment:
@aanm ah, must have left this on a different machine. It was marked as outdated so I just assumed it was addressed since Github automatically collapses these. Will follow up. Thanks!